packages/swapper/src/swappers/ButterSwap/swapperApi/getTradeQuote.ts

```diff
@@ -0,0 +1,232 @@
+      }),
+    )
+  }
+
+  // Call ButterSwap /swap API to get calldata and contract info
+  const buildTxResult = await getBuildTx(
+    route.hash,
+    slippage,
+    sendAddress, // from (source chain address)
+    receiveAddress, // receiver (destination chain address)
+  )
+
+  if (buildTxResult.isErr()) return Err(buildTxResult.unwrapErr())
+  const buildTxResponse = buildTxResult.unwrap()
+  if (!isBuildTxSuccess(buildTxResponse)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `[getTradeQuote] /swap failed: ${buildTxResponse.message}`,
+        code: TradeQuoteError.QueryFailed,
+      }),
+    )
+  }
+  const buildTx = buildTxResponse.data[0]
+  if (!buildTx) {
+    return Err(
+      makeSwapErrorRight({
+        message: '[getTradeQuote] No buildTx data returned',
+        code: TradeQuoteError.QueryFailed,
+      }),
+    )
+  }
+
+  // Fee asset for network/protocol fees
+  const feeAsset = _deps.assetsById[feeAssetId]
+  if (!feeAsset) {
+    return Err(
+      makeSwapErrorRight({
+        message: `[getTradeQuote] Fee asset not found for chainId ${sellAsset.chainId}`,
+        code: TradeQuoteError.UnsupportedChain,
+      }),
+    )
+  }
+
+  // Map gasFee.amount to networkFeeCryptoBaseUnit using fee asset precision
+  const networkFeeCryptoBaseUnit = bnOrZero(route.gasFee?.amount).gt(0)
+    ? toBaseUnit(route.gasFee.amount, feeAsset.precision)
+    : '0'
+
+  const nativeFeeBaseUnit = bnOrZero(route.swapFee.nativeFee).gt(0)
+    ? toBaseUnit(route.swapFee.nativeFee, feeAsset.precision)
+    : '0'
+  const tokenFeeBaseUnit = bnOrZero(route.swapFee.tokenFee).gt(0)
+    ? toBaseUnit(route.swapFee.tokenFee, sellAsset.precision)
+    : '0'
+
+  let protocolFees: Record<string, ProtocolFee> | undefined = undefined
+  const hasNativeFee = bnOrZero(nativeFeeBaseUnit).gt(0)
+  const hasTokenFee = bnOrZero(tokenFeeBaseUnit).gt(0)
+
+  if (hasNativeFee && hasTokenFee && feeAssetId === sellAsset.assetId) {
+    // If both fees are for the same asset, sum them
+    protocolFees = {
+      [feeAssetId]: {
+        amountCryptoBaseUnit: bnOrZero(nativeFeeBaseUnit).plus(tokenFeeBaseUnit).toString(),
+        requiresBalance: true,
+        asset: feeAsset,
+      },
+    }
+  } else {
+    protocolFees = {}
+    if (hasNativeFee) {
+      protocolFees[feeAssetId] = {
+        amountCryptoBaseUnit: nativeFeeBaseUnit,
+        requiresBalance: true,
+        asset: feeAsset,
+      }
+    }
+    if (hasTokenFee) {
+      protocolFees[sellAsset.assetId] = {
+        amountCryptoBaseUnit: tokenFeeBaseUnit,
+        requiresBalance: true,
+        asset: sellAsset,
+      }
+    }
+    if (Object.keys(protocolFees).length === 0) protocolFees = undefined
+  }
+
+  // Calculate rate as lastHop.totalAmountOut / srcChain.totalAmountIn (in base units)
+  const inputAmount = bnOrZero(route.srcChain.totalAmountIn)
+  const outputAmount =
+    route.dstChain?.totalAmountOut ??
+    route.bridgeChain?.totalAmountOut ??
```

Comment:
not sure about this one, from testing things out a bit at runtime, this seems to be the intermediary chain for cross-chain, so we should remove it I reckon? 
i.e :
- same chain will have final min amount out on `srcChain` (there is no `dstChain`, so we'll fallback to it)
- cross-chain will have final min amount out on `dstChain` so we'll hit the first
 - we should never hit the second one